syscall build error on GOOS=darwin

Hi everyone,

when I import other than the gopherjs library to my go file,  and when I ran the ` "gopherjs build"` cmd, it  throws errors likes those, but when I import only gopherjs it doesn't give me anything error: 
`
/usr/local/opt/go/libexec/src/syscall/syscall_darwin.go:118:1: //go:linkname local symbol "libc_getfsstat" is not found in the current source file
/usr/local/opt/go/libexec/src/syscall/syscall_darwin.go:151:1: //go:linkname local symbol "libc_setattrlist" is not found in the current source file
/usr/local/opt/go/libexec/src/syscall/syscall_darwin.go:279:1: //go:linkname local symbol "libc_fdopendir" is not found in the current source file
/usr/local/opt/go/libexec/src/syscall/syscall_darwin_amd64.go:59:1: //go:linkname local symbol "libc_sendfile" is not found in the current source file
/usr/local/opt/go/libexec/src/syscall/zsyscall_darwin_amd64.go:23:1: //go:linkname local symbol "libc_getgroups" is not found in the current source file
/usr/local/opt/go/libexec/src/syscall/zsyscall_darwin_amd64.go:38:1: //go:linkname local symbol "libc_setgroups" is not found in the current source file
/usr/local/opt/go/libexec/src/syscall/zsyscall_darwin_amd64.go:54:1: //go:linkname local symbol "libc_wait4" is not found in the current source file
/usr/local/opt/go/libexec/src/syscall/zsyscall_darwin_amd64.go:70:1: //go:linkname local symbol "libc_accept" is not found in the current source file
/usr/local/opt/go/libexec/src/syscall/zsyscall_darwin_amd64.go:85:1: //go:linkname local symbol "libc_bind" is not found in the current source file
/usr/local/opt/go/libexec/src/syscall/zsyscall_darwin_amd64.go:100:1: //go:linkname local symbol "libc_connect" is not found in the current source file
:0:0: too many errors
`

My go file :

`
```
package main

import (
"github.com/ava-labs/avalanchego/utils/formatting"

"github.com/gopherjs/gopherjs/js"
)

func Try() string { return "hello world"}
 
func main() {
	js.Global.Get("module").Get("exports").Set("Try", Try)
}
```
`


why is this happening?  how do I solve this error?

thank you

--- a/build/build.go
+++ b/build/build.go
@@ -175,10 +175,11 @@ func importWithSrcDir(bctx build.Context, path string, srcDir string, mode build
 	switch path {
 	case "os":
 		pkg.GoFiles = excludeExecutable(pkg.GoFiles) // Need to exclude executable implementation files, because some of them contain package scope variables that perform (indirectly) syscalls on init.
-		// Prefer dirent_js.go version, since it targets a similar environment to
-		// ours. Arguably this file should be excluded by the build tags (see
-		// https://github.com/gopherjs/gopherjs/issues/693).
-		pkg.GoFiles = exclude(pkg.GoFiles, "dirent_linux.go")
+		// Prefer the dirent_${GOOS}.go version, to make the build pass on both linux
+		// and darwin.
+		// In the long term, our builds should produce the same output regardless
+		// of the host OS: https://github.com/gopherjs/gopherjs/issues/693.
+		pkg.GoFiles = exclude(pkg.GoFiles, "dirent_js.go")
 	case "runtime":
 		pkg.GoFiles = []string{} // Package sources are completely replaced in natives.
 	case "runtime/internal/sys":
@@ -194,6 +195,16 @@ func importWithSrcDir(bctx build.Context, path string, srcDir string, mode build
 	case "crypto/rand":
 		pkg.GoFiles = []string{"rand.go", "util.go"}
 		pkg.TestGoFiles = exclude(pkg.TestGoFiles, "rand_linux_test.go") // Don't want linux-specific tests (since linux-specific package files are excluded too).
+	case "crypto/x509":
+		// GopherJS doesn't support loading OS root certificates regardless of the
+		// OS. The substitution below allows to avoid build dependency on Mac OS
+		// implementation, which won't be used anyway.
+		//
+		// Just like above, https://github.com/gopherjs/gopherjs/issues/693 is
+		// probably the best long-term option.
+		pkg.GoFiles = include(
+			exclude(pkg.GoFiles, fmt.Sprintf("root_%s.go", bctx.GOOS)),
+			"root_unix.go", "root_js.go")
 	}
 
 	if len(pkg.CgoFiles) > 0 {
@@ -247,6 +258,12 @@ Outer:
 		s = append(s, f)
 	}
 	return s
+}
+
+func include(files []string, includes ...string) []string {
+	files = exclude(files, includes...) // Ensure there won't be duplicates.
+	files = append(files, includes...)
+	return files
 }
 
 // ImportDir is like Import but processes the Go package found in the named

--- a/compiler/linkname.go
+++ b/compiler/linkname.go
@@ -101,6 +101,15 @@ func parseGoLinknames(fset *token.FileSet, pkgPath string, file *ast.File) ([]Go
 
 		obj := file.Scope.Lookup(localName)
 		if obj == nil {
+			if pkgPath == "syscall" {
+				// Syscall uses go:cgo_import_dynamic pragma to import symbols from
+				// dynamic libraries when build with GOOS=darwin, which GopherJS doesn't
+				// support. Silently ignore such directives.
+				//
+				// In the long term https://github.com/gopherjs/gopherjs/issues/693 is a
+				// preferred solution.
+				return nil
+			}
 			return fmt.Errorf("//go:linkname local symbol %q is not found in the current source file", localName)
 		}
 

--- a/compiler/natives/src/syscall/syscall_darwin.go
+++ b/compiler/natives/src/syscall/syscall_darwin.go
@@ -20,10 +20,6 @@ func funcPC(f func()) uintptr {
 		return SYS_CHDIR
 	case js.InternalObject(libc_rmdir_trampoline):
 		return SYS_RMDIR
-	case js.InternalObject(libc___getdirentries64_trampoline):
-		return SYS_GETDIRENTRIES64
-	case js.InternalObject(libc_getattrlist_trampoline):
-		return SYS_GETATTRLIST
 	case js.InternalObject(libc_symlink_trampoline):
 		return SYS_SYMLINK
 	case js.InternalObject(libc_readlink_trampoline):

